### ISSUE TITLE
Wrong result for `setcontent page = "pages" where {'id': 16}`

### ISSUE THREAD

**Issue by bobdenotter** [bolt/four#283]
If i go to `/page/16`, with page # 16 being a `published` page, and I edit `record.twig` to look like this: 

```twig

{% block main %}

    <h1>{{ record.title }}</h1>

    {{ dump(record.title) }}
    {{ dump(record.id) }}

    {% setcontent page = "pages" where {'id': 16} %}
    {{ dump(page|first.title) }}
    {{ dump(page|first.id) }}
```

I would expect the two dumps to look similar. This is the result: 

![screenshot 2019-01-21 at 19 55 18](https://user-images.githubusercontent.com/1833361/51493723-8094ce80-1db6-11e9-8906-ed815719d220.png)

So, they give a different result, coming from a different record, and the second one has an additional `, UTF-8`, like it's passed through `Twig_Markup` twice.

**Comment by bobdenotter** [comment]
Partial fix (for the `, UTF-8` issue) coming up.

About the other, it seems to "skip" the `where`: 

```
{% setcontent page = "pages" where { 'id': 16 } printquery %}

=> SELECT content FROM Bolt\Entity\Content content WHERE content.status = :status_1 
```

**Comment by xiaohutai** [comment]
Fixed in #286
